### PR TITLE
[Merged by Bors] - Remove orphaned resources

### LINKED ISSUE #72: Handle stale information/clean up stale resources
Currently our operators will not act on removed information from the CR in some/most/all cases.

One example:
HBase operator has three roles (master, regionServer, restServer). If I create a HBase server CR with a restServer component and then remove it later (entirely, not setting replicas to 0) our operator will not clean up the STS that belongs to this role.

## Proposed solution

Since operator-rs 0.24.0 there is a `ClusterResource` struct which is used to find and track Kubernetes resources belonging to a Cluster object. An example of its usage can be found in the Superset Operator: https://github.com/stackabletech/superset-operator/blob/main/rust/operator-binary/src/superset_controller.rs#L241

## Acceptance criteria

- [x] This is done when all stale Kubernetes resources are cleaned up. A resource becomes stale when it's not part of the current cluster definition anymore.
- [x] The `AirflowDB` resource is excluded from this recycling process.
- [x] Also referenced `ConfigMaps` and `PersistenVolumeClaims` are not deleted.
- [x] There is at least one test.
- [ ] Documentation on operator implementation is updated with information regarding handling of stale resources.
- [x] Upgrade to the latest `operator-rs` version (`0.25` at the moment)

NOTE: This is part of an epic (https://github.com/stackabletech/issues/issues/203) and _might_ not apply to this operator. If that is the case please comment on this issue and just close it. This issue was created as part of a special bulk creation operation.

### PR DESCRIPTION
**Pull request by adwk67**
# Description

Implement removal of orphaned resources. Closes #72.
Jenkins: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/airflow-operator-it-custom/44/

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Review by fhennig — CHANGES_REQUESTED** [review]
From reading through, I just foud one thing

File: rust/crd/src/lib.rs [hunk 1]
```diff
@@ -17,6 +17,7 @@ use stackable_operator::schemars::{self, JsonSchema};
 use strum::{Display, EnumIter, EnumString};
 
 pub const APP_NAME: &str = "airflow";
+pub const CONTROLLER_NAME: &str = "airflow-controller";
```
[the remark this answers is not in the document]
Comment (adwk67) [reply rust/crd/src/lib.rs:20]: My understanding from the discussion in the hbase repo yesterday, was that the controller name is correct, as there is more than one managed by the operator.

[the remark this answers is not in the document]
Comment (fhennig) [reply rust/crd/src/lib.rs:20]: Ok

**Comment by fhennig** [comment]
Okay, I'm fine with making the docs about this some other time, I've added a comment to this ticket https://github.com/stackabletech/documentation/issues/282

**Review by fhennig — APPROVED** [review]

**Comment by adwk67** [comment]
bors merge

**Pull request closed without merging**